### ISSUE TITLE
Uknown extension vscode.ipynb

### ISSUE THREAD

**Issue by thistlillo** [microsoft/vscode-jupyter#9235]
All of a sudden (yesterday I was using the Python notebooks on a remote machine, then disconnected and today I connected again) I cannot even open ipynb files on VS Code.

I did not install other extension, nor did I upgrade VSC.

```
Version: 1.64.2 (user setup)
Commit: f80445acd5a3dadef24aa209168452a3d97cc326
Date: 2022-02-09T22:02:28.252Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043
```
The remote machine is a kubernetes pod running linux ubuntu 20.04.

I searched but could not find any answer.

**Comment by amunger** [comment]
> python.setInterpreter not found
This sounds like your python extension isn't configured correctly. Can you try reinstalling that extension? Also, what version of that extension are you using?

**Comment by amunger** [comment]
also, can you make sure that the Jupyter extension you are troubleshooting is the one on the remote machine, not the local instance of VS code.

**Comment by thistlillo** [comment]
Hello Aaron and thank you for your support. The remote Python environment
was working fine during the last connection. However it is possible, even
if I did not notice anything, that VSC got updated automatically, it seems
there was a recent update to the February 2022. I have found references to
the same issue where the cause was the update of the app.

After sending this message, i uninstalled the Python and the Jupyter
extensions by Microsoft both remotely and locally, removed the remote
.vscode/ server data and reinstalled. Now it is working fine again.

I have also disabled the automatic updates.

Il ven 4 mar 2022, 21:46 Aaron Munger ***@***.***> ha scritto:

**Comment by amunger** [comment]
thanks for the update @thistlillo. So are you on the latest release for everything now (VSC, jupyter, Python), or did you have to revert back to an older version to get it working?

**Comment by thistlillo** [comment]
Actually there is a bigger problem here, at least on my installation.

I have now the latest version of everything (VSC, Python and Jupyter extensions by Microsoft) and every time I close the remote connection, the python setup breaks.

For example, two hours ago I was working with python files and jupyter notebooks on the remote hosts, now I have just reconnected and nothing is working any more.

Jupyter notebooks are not even opened, Python .py files are read as normal txt files. Indeed, if I try and set a Python interpreter I get this message :

![image](https://user-images.githubusercontent.com/6717655/156880811-1a6e35f9-d52c-40f7-8da0-4a110266efbc.png)

I do not know what update broke everything, but VSC has become unusuable.

**Comment by thistlillo** [comment]
Yes, again today. After I disconnect from the remote host the Python setup gets messed up:

![image](https://user-images.githubusercontent.com/6717655/156912936-7dda283f-4911-4071-89ec-3183fde0d8e6.png)

May be this is due to some other extensions, however I did not update anything, if not VSC that got updated automatically.

**Comment by thistlillo** [comment]
Solved, the issue was caused by an extension for syncing Settings. Despite having already disabled the syncing of potentially conflicting settings, after the update of VSC this extension was causing issues.

This experience, like others, make all the "extensions" idea, even if beautiful, quite fragile and annoying: I think that currently only extensions from major players can really be used in a production environment. It is not reasonable to pretend users to make an extensive check of all the extensions at every update of VSC.
